### ISSUE TITLE
ECS Autoscaling Role Drift After Apply

### ISSUE THREAD

**Issue by sps-software** [cn-terraform/terraform-aws-ecs-fargate-service#14]
Hey,

Something a bit odd is happening each time after creating or updating a service. Each time I apply after initially bringing it up, the autoscaling role has been changed to the AWS managed ECS autoscaling role. So I get this plan output every time:
```
~ resource "aws_appautoscaling_target" "scale_target" {
        id                 = "service/artspot/hello-world-service"
        max_capacity       = 5
        min_capacity       = 1
        resource_id        = "service/artspot/hello-world-service"
      ~ role_arn           = "arn:aws:iam::XXXXXXXXXXXX:role/aws-service-role/ecs.application-autoscaling.amazonaws.com/AWSServiceRoleForApplicationAutoScaling_ECSService" -> "arn:aws:iam::XXXXXXXXXXXX:role/hello-world-ecs-autoscale-role"
        scalable_dimension = "ecs:service:DesiredCount"
        service_namespace  = "ecs"
    }
```

Might not be anything you can do about it. Anyway, great module.

**Comment by jnonino** [comment]
Hi, I have made a quick change in how the policy is set. It is on version [2.0.4](https://registry.terraform.io/modules/cn-terraform/ecs-fargate-service/aws/2.0.4). I'm not sure that fixes the issue but it was something to improve. Later I will check more in detail if I can find something about it

**Comment by sps-software** [comment]
That seems to be resulting in this error:
```
Error: Error attaching policy arn:aws:iam::aws:policy/aws-service-role/AWSApplicationAutoscalingECSServicePolicy to IAM Role hello-world-ecs-autoscale-role: PolicyNotAttachable: Cannot attach a Service Role Policy to a Customer Role.
        status code: 400, request id: ...

  on .terraform/modules/ecs-fargate-service/terraform-aws-ecs-fargate-service-2.0.4/autoscaling.tf line 9, in resource "aws_iam_role_policy_attachment" "ecs_autoscale_role_policy_attach":
   9: resource "aws_iam_role_policy_attachment" "ecs_autoscale_role_policy_attach" {
```

**Comment by jnonino** [comment]
Reverted

**Comment by jnonino** [comment]
Released in version [2.0.9](https://registry.terraform.io/modules/cn-terraform/ecs-fargate-service/aws/2.0.9)
